src/main/java/net/openhft/chronicle/wire/WireMarshaller.java

```diff
@@ -2022,6 +2022,8 @@ else if (type == Set.class)
             componentType = extractClass(computeActualTypeArguments(Collection.class, field)[0]);
             if (componentType != Object.class) {
                 isLeaf = !Throwable.class.isAssignableFrom(componentType)
+                        // Don't recurse into the same class
+                        && !componentType.equals(field.getDeclaringClass())
```

Comment:
Wasn't sure if this should be !componentType.isAssignmentFrom(field.getDeclaringClass())

Reply:
I think equality is correct as the test whereby the types were not the same but one assignable from the other, seemed to work OK